### PR TITLE
feature(Array): add merge function

### PR DESCRIPTION
**Pull request by anta-semenov**

### CODE REVIEW

File: src/array.ts [hunk 1]
```diff
@@ -416,3 +417,31 @@ export const takeWhile = <T>(inputArray: T[], filterFunction: (element: T, index
...
+      result.push(secondArray[secondArrayIndex])
+      secondArrayIndex++
+    }
+  }
+
+  return result.unique()
+}
+
+if (!Array.prototype.mergeWith) {
+  Array.prototype.mergeWith = function<T>(seccondArray: T[]): T[] {
```
Comment (lucjac) [inline src/array.ts]: typo seccond?

**Comment by lucjac** [comment]
btw since it seems like it would take any type, does this make sense for e.g. arrays of booleans? if would probably remove all entries except two. Maybe restrict the input types more?

File: src/array.ts (hunk 1 above)
Comment (anta-semenov) [reply src/array.ts]: oops, thank you

**Comment by anta-semenov** [comment]
Good point, probably it makes sense to use it only for strings/numbers arrays

**Pull request merged**